godtossing/Pos/Ssc/GodTossing/Arbitrary.hs

```diff
@@ -103,18 +107,22 @@ commitmentsAndOpenings =
     vssPk = toVssPublicKey $ deterministicVssKeyGen "ababahalamaha"
 {-# NOINLINE commitmentsAndOpenings #-}
 
-
 instance Arbitrary CommitmentOpening where
     arbitrary = elements commitmentsAndOpenings
+    shrink = genericShrink
 
 instance Nonrepeating CommitmentOpening where
     nonrepeating n = sublistN n commitmentsAndOpenings
 
 instance Arbitrary Commitment where
     arbitrary = coCommitment <$> arbitrary
+    shrink Commitment {..} = [ Commitment { commShares = shrunkShares, .. }
```

Comment:
Actually, you should check invarians on all data types. If you risk breaking it, it's better to have no shrinking (otherwise you can't rely on the invariant in tests).